packages/flutter_tools/lib/src/flutter_plugins.dart

```diff
@@ -382,8 +380,8 @@ Future<void> _writeAndroidPluginRegistrant(FlutterProject project, List<Plugin>
       templateContext['needsShim'] = false;
       // If a plugin is using an embedding version older than 2.0 and the app is using 2.0,
       // then add shim for the old plugins.
-      for (final Map<String, dynamic> plugin in androidPlugins) {
-        if (plugin['supportsEmbeddingV1'] as bool && !(plugin['supportsEmbeddingV2'] as bool)) {
+      for (final Map<String, Object?> plugin in androidPlugins) {
+        if ((plugin['supportsEmbeddingV1'] as bool?) == true && (plugin['supportsEmbeddingV2'] as bool?) != true) {
```

Comment:
Can you double check me here?

Reply:
I think this would read a lot better if you pulled these out into locals, otherwise the logic looks correct:

```
final bool supportsEmbeddingV1 = plugin['supportsEmbeddingV1'] as bool? == true;
final bool supportsEmbeddingV2 = plugin['supportsEmbeddingV2'] as bool? == true;
if (supportsEmbeddingV1 && !supportsEmbeddingV1) { ... }
```

Reply:
Good idea.